sdk/identity/Azure.Identity/src/ManagedIdentityResponseClassifier.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+
+namespace Azure.Identity
+{
+    internal class ManagedIdentityResponseClassifier : ResponseClassifier
+    {
+        public override bool IsRetriableResponse(HttpMessage message)
+        {
+            return message.Response.Status switch
+            {
+                404 => true,
+                410 => true,
+                502 => false,
+                _ => base.IsRetriableResponse(message)
```

Comment:
Out-of-scope of this PR, since it is just a move:
Can you share any doc or reference that explains why we treat 404/410 as retriable, and 502 as non-retriable as an extension to the base behavior?

Might be worth adding as a comment here.

Reply:
https://learn.microsoft.com/en-us/azure/virtual-machines/instance-metadata-service?tabs=windows#errors-and-debugging

Reply:
added a comment